### PR TITLE
add qna maker extension

### LINKED ISSUE #4290: Add qna maker extension
## Version
What package version of the SDK are you using.

## Describe the bug
Similar to LuisExtentions, we need to use QnAMakerExtentions to obtain some config resources. For example, we need to trace the qnamaker.setting.development.westus.json file in the example bot folder.
## Expected behavior
Give a clear and concise description of what you expected to happen.

## Additional context

[bug]

### PR DESCRIPTION
**Pull request by coldplaying42**
Fixes #4290 

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtension.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        /// Setup configuration to utilize the settings file generated by bf qnamaker:build.
+        /// </summary>
+        /// <remarks>
+        /// This will pick up --environment as the environment to target.  If environment is Development it will use the name of the logged in user.
+        /// This will pick up --root as the root folder to run in.  
+        /// </remarks>
+        /// <param name="builder">Configuration builder to modify.</param>
+        /// <param name="botRoot">bot Root.</param>
+        /// <param name="qnaRegion">qnaRegion.</param>
+        /// <param name="environment">enviroment.</param>
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtension.cs:28]: Please add useful information.

File: tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerSettingTests.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Bot.Builder.AI.QnA.Tests
+{
+    [TestClass]
+    public class QnAMakerSettingTests
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerSettingTests.cs:11]: Please rename the test class to QnAMakerSettingsTests

File: tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerSettingTests.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            var builder = new ConfigurationBuilder();
+            var config = builder.Build();
+            var botRoot = config.GetValue<string>("root") ?? ".";
+            var region = config.GetValue<string>("region") ?? "westus";
+            var environment = config.GetValue<string>("environment") ?? "development";
+
+            builder.UseQnAMakerSettings(botRoot, region, environment);
+
+            var source = builder.Sources[1] as Microsoft.Extensions.Configuration.Json.JsonConfigurationSource;
+            Assert.AreEqual(source.Path, "qnamaker.settings.development.westus.json");
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerSettingTests.cs:27]: The expected string should be the first parameter.

File: tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerSettingTests.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace Microsoft.Bot.Builder.AI.QnA.Tests
+{
+    [TestClass]
+    public class QnAMakerSettingTests
+    {
+        [TestMethod]
+        [TestCategory("AI")]
+        [TestCategory("QnAMaker")]
+        public void QnAMakerSetting()
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerSettingTests.cs:16]: A better name for this test should suggest what you are testing.

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtension.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Bot.Builder.AI.QnA
+{
+    /// <summary>
+    /// Extension methods for QnA.
+    /// </summary>
+    public static class QnAMakerExtension
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtension.cs:16]: Please rename the class to IConfigurationBuilderExtensions.

Comment (coldplaying42) [reply libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtension.cs:16]: I follow the name of luisExtension.cs. Do I have to rename it?

Comment (gabog) [reply libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtension.cs:16]: Yes please, the luisExtensions file is wrongly named too. In general, extension methods should be placed in files called _TypeName_Extensions.cs, there `TypeName`, is the name of the type that the extension methods apply to. This makes it easier to spot them in the source tree. 

We haven't followed this naming convention consistently but we should.

**Pull request merged**